programs/sbf/benches/bpf_loader.rs

```diff
@@ -337,23 +337,26 @@ fn clone_regions(regions: &[MemoryRegion]) -> Vec<MemoryRegion> {
     unsafe {
         regions
             .iter()
-            .map(|region| match region.state.get() {
-                MemoryState::Readable => MemoryRegion::new_readonly(
-                    slice::from_raw_parts(region.host_addr.get() as *const _, region.len as usize),
-                    region.vm_addr,
-                ),
-                MemoryState::Writable => MemoryRegion::new_writable(
-                    slice::from_raw_parts_mut(
-                        region.host_addr.get() as *mut _,
-                        region.len as usize,
-                    ),
-                    region.vm_addr,
-                ),
-                MemoryState::Cow(id) => MemoryRegion::new_cow(
-                    slice::from_raw_parts(region.host_addr.get() as *const _, region.len as usize),
-                    region.vm_addr,
-                    id,
-                ),
+            .map(|region| {
+                let mut new_region = if region.writable.get() {
+                    MemoryRegion::new_writable(
+                        slice::from_raw_parts_mut(
+                            region.host_addr.get() as *mut _,
+                            region.len as usize,
+                        ),
+                        region.vm_addr,
+                    )
+                } else {
+                    MemoryRegion::new_readonly(
+                        slice::from_raw_parts(
+                            region.host_addr.get() as *const _,
+                            region.len as usize,
+                        ),
+                        region.vm_addr,
+                    )
+                };
+                new_region.cow_callback_payload = region.cow_callback_payload;
```

Comment:
I just noticed the payload could be an optional payload in `new_readonly` and `new_writable`, but I'll spare you the job of fixing two repos and doing another release.